### PR TITLE
[SYCLomatic][intercept-build] Refine libear's implementation to fix the failure of 4 lit tests related to intercept-build tool in corner case

### PR DESCRIPTION
**Pull request by tomflinda**
The related cases are as follows:
1. dpct/cmp-cmds-linker-entry-src-files/one.cu
2. dpct/dpct-intercept-build/main.cpp
3. dpct/gen_build_script2/hello.cu
4. dpct/gen_build_script3/hello.cu

Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

**Review by ShengchenJ — APPROVED** [review]

File: clang/tools/scan-build-py/lib/libear/ear.c [hunk 1]
```diff
@@ -487,28 +487,39 @@ static int call_eaccess(const char *pathname, int mode) {
...
 }
 
-int eaccess(const char *pathname, int mode) {
-  int ret = call_eaccess(pathname, mode);
-  if (ret == 0) {
-    return 0;
-  }
+int is_nvcc_available(void) {
+  char *value_compile = getenv("INTERCEPT_COMPILE_PATH");
+  char *value_env = getenv("IS_INTERCEPT_COMPILE_PATH_FROM_ENV_PATH");
```
Comment (zhimingwang36) [inline clang/tools/scan-build-py/lib/libear/ear.c]: is_compiler_exported?

File: clang/tools/scan-build-py/lib/libear/ear.c [hunk 2]
```diff
@@ -487,28 +487,39 @@ static int call_eaccess(const char *pathname, int mode) {
   return result;
 }
 
-int eaccess(const char *pathname, int mode) {
-  int ret = call_eaccess(pathname, mode);
-  if (ret == 0) {
-    return 0;
-  }
+int is_nvcc_available(void) {
+  char *value_compile = getenv("INTERCEPT_COMPILE_PATH");
```
Comment (zhimingwang36) [inline clang/tools/scan-build-py/lib/libear/ear.c]: compiler?

**Review by zhimingwang36 — APPROVED** [review]
